interfaces/Portalicious/src/app/services/auth.service.ts

```diff
@@ -207,4 +185,22 @@ export class AuthService {
       }),
     );
   }
+
+  async processAzureCallback() {
+    try {
+      const currentUser = await this.queryClient.fetchQuery(
+        this.userApiService.getCurrent()(),
+      );
+      setUserInLocalStorage(currentUser.user);
+    } catch (error) {
+      // TODO: AB#31489 Return a more generic endpoint from the back-end
+      if (error instanceof HttpErrorResponse) {
+        const errorMessage = get(error.error, 'message') as string | undefined;
+        if (errorMessage) {
+          this.authError.set(errorMessage);
+        }
+      }
+      throw error;
+    }
+  }
```

Comment:
```suggestion
  async refreshUser() {
    const currentUser = await this.queryClient.fetchQuery(
      this.userApiService.getCurrent()(),
    );
    setUserInLocalStorage(currentUser.user);
  }
```

Then the error catching / handling happens in the component. And I think this way we might be able to avoid setting `authError` in the auth service too.